admin/gv_mail.php

```diff
@@ -4,421 +4,382 @@
-      $message .= "\n\n" . TEXT_GV_WORTH  . $currencies->format($_POST['amount']) . "\n\n";
-      $message .= TEXT_TO_REDEEM;
-      $message .= TEXT_WHICH_IS . ' ' . $id1 . ' ' . TEXT_IN_CASE . "\n\n";
-
-      $html_msg['GV_WORTH']  = TEXT_GV_WORTH;
-      $html_msg['GV_AMOUNT']  = $currencies->format($_POST['amount']);
-      $html_msg['GV_REDEEM'] = TEXT_TO_REDEEM . TEXT_WHICH_IS . ' <strong>' . $id1 . '</strong> ' . TEXT_IN_CASE . "\n\n";
-
-      if (SEARCH_ENGINE_FRIENDLY_URLS == 'true') {
-        $message .= HTTP_CATALOG_SERVER  . DIR_WS_CATALOG . 'index.php/gv_redeem/gv_no/'.$id1 . "\n\n";
-        $html_msg['GV_CODE_URL']  = '<a href="' . HTTP_CATALOG_SERVER . DIR_WS_CATALOG . 'index.php/gv_redeem/gv_no/'.$id1.'">' .TEXT_CLICK_TO_REDEEM . '</a>'. "&nbsp;";
-      } else {
-        $message .= HTTP_CATALOG_SERVER  . DIR_WS_CATALOG . 'index.php?main_page=gv_redeem&gv_no='.$id1 . "\n\n";
-        $html_msg['GV_CODE_URL']  =  '<a href="'. HTTP_CATALOG_SERVER  . DIR_WS_CATALOG . 'index.php?main_page=gv_redeem&gv_no='.$id1 .'">' .TEXT_CLICK_TO_REDEEM . '</a>' . "&nbsp;";
-      }
-      $message .= TEXT_OR_VISIT . HTTP_CATALOG_SERVER  . DIR_WS_CATALOG  . TEXT_ENTER_CODE . "\n\n";
-      $html_msg['GV_CODE_URL']  .= TEXT_OR_VISIT .  '<a href="'.HTTP_CATALOG_SERVER  . DIR_WS_CATALOG.'">' . STORE_NAME . '</a>' . TEXT_ENTER_CODE;
-
-      $html_msg['EMAIL_MESSAGE_HTML'] = zen_db_prepare_input($_POST['message_html']);
-      $html_msg['EMAIL_FIRST_NAME'] = ''; // unknown, since only an email address was supplied
-      $html_msg['EMAIL_LAST_NAME']  = ''; // unknown, since only an email address was supplied
-
-      // disclaimer
-      $message .= "\n-----\n" . sprintf(EMAIL_DISCLAIMER, STORE_OWNER_EMAIL_ADDRESS) . "\n\n";
-
-      //Send the emails
-      zen_mail('Friend', $_POST['email_to'], $subject , $message, $from, $from, $html_msg, 'gv_mail');
-      $recip_count++;
-      if (SEND_EXTRA_GV_ADMIN_EMAILS_TO_STATUS== '1' and SEND_EXTRA_GV_ADMIN_EMAILS_TO != '') {
-        zen_mail('', SEND_EXTRA_GV_ADMIN_EMAILS_TO, SEND_EXTRA_GV_ADMIN_EMAILS_TO_SUBJECT . ' ' . $subject, $message, $from, $from, $html_msg, 'gv_mail_extra');
-      }
-
-      // Now create the coupon main entry
-      $insert_query = $db->Execute("insert into " . TABLE_COUPONS . "
-                                    (coupon_code, coupon_type, coupon_amount, date_created)
-                                    values ('" . zen_db_input($id1) . "', 'G', '" . zen_db_input($_POST['amount']) . "', now())");
+                $message = EMAIL_SALUTATION . ' ' . $row['customers_firstname'] . ' ' . $row['customers_lastname'] . ',' . "\n\n";
+                $html_msg['EMAIL_SALUTATION'] = EMAIL_SALUTATION;
+                $html_msg['EMAIL_FIRST_NAME'] = $row['customers_firstname'];
+                $html_msg['EMAIL_LAST_NAME'] = $row['customers_lastname'];
+
+                $message .= zen_db_prepare_input($_POST['message']);//steve

```

Comment:
Is this comment suggesting it's a temporary line, or just left as an artifact from prior code?

Reply:
No, it's meant to tell me I've added that bit...should have removed it.